## Capítulo 1.1.: Modelos estadísticos para control de calidad

### Caso 1.1.1: Análisis de capacidad de proceso

In [8]:
import numpy as np
import math
import scipy.stats as st

#### Ejemplo 6:

Una reconocida institución de cursos en línea se encuentra ante una incertidumbre pues su operador de redes e internet les notificó la semana pasada formalmente que dejará de funcionar en tres (3) meses. Al rescate ha surgido una nueva empresa en el mercado proveniente de España que le hizo una oferta económicamente atractiva a la institución, sin embargo el CEO de la institución les advirtió lo siguiente a través de un correo:

    "Respetados Operador Español,

    Estamos muy agradecidos por la oportuna oferta, pero dejeme advertiles que la velocidad de transferencia de datos promedio que requiere esta institución es de 375 Mbps, y dicho valor puede discrepar máximo en 20 Mbps.*

    Llevamos trabajando más de quince (15) años de la mano de nuestro actual operador. Ustedes entenderán que esto se trata de establecer una nueva relación de negocios (y de confianza) que deberemos emprender, así que estudiaremos estrictamente las opciones que tengamos a la mano antes de tomar cualquier decisión, pues quienes contraten con nosotros deberán cumplir estrictamente con nuestras especificaciones en todo momento como lo venía haciendo nuestro actual y próximamente antiguo operador.*

    Agradezco de antemano la atención prestada,

    Cordialmente,
    Jose Mantilla Solano
    CEO" 

Luego de recibir el correo electrónico, el Gerente del Operador de Red respondió lo siguiente:

    "Respetado Jose Mantilla Solano,
     CEO Institución.
    
    No os preocupeís estimado! Por el contrario, sepa usted que nuestra velocidad promedio de transferencia de datos para clientes corporativos es de 370 Mbps acompañado de una desviación típica de 11 Mbps.

    Somos reconocidos en España por prestar un excelente servicio de calidad y contamos con canales de soporte al cliente las 24 horas del día. Nos gustaría que considerara responder a nuestra oferta en el más corto de los términos, pero sabemos muy bien que las relaciones de confianza tardan años en consolidarse. Por esta razón, le aseguro que los parámetros que le ofrezco serán más que suficientes para soportar e impulsar su negocio.

    Espero atento a su respuesta.

    Rafael Cervantez Silva,
    Gerente Comercial"

Ante tal respuesta, el CEO de la Institución se mostró suspicaz por lo que llamó a su empleado experto en Aseguramiento y Control de la Calidad quien decidió hacer un análisis de la capacidad del proceso utilizando los parámetros que les compartió el gerente comercial.

**a) Estimar la Capacidad Potencial (Cp):**

R/ Se sabe que el índice de capacidad del proceso Cp esta dado por:

$$\rightarrow C_{p}=\frac{T}{6\sigma'}$$

Y:
$$\rightarrow T = 2\nabla = LSE - LIE$$

Donde:
* $\nabla$ es la discrepancia y de ahora en adelante se le asignará la letra 'D', 
* $\sigma'$ que es la desviación estándar se le asignará la letra 's',y 
* 'T' es la tolerancia. 

A continuación se procederá determinar la capacidad del proceso:

In [19]:
D = 20
VN = 375
LIE = VN - D
LSE = VN + D
T = 2*D
s = 11 # Se toma s como el mejor estimador de sigma.

Cp = T/(6*s)
print(Cp)

0.6060606060606061


Dado que $C_{p} < 0.67$ entonces esto indica que el proceso tiene una muy pobre capacidad de ofrecer una velocidad de transferencia dentro de las especificaciones, y según los criterios se trata de un proceso no adecuado para el trabajo, por lo que requiere modificaciones serias.

**b) Estimar la Capacidad Real (Cpk):**

R/ Cuando el proceso no está centrado, se emplea el **índice de capacidad unilateral** o **índice de capacidad real (Cpk)**:

$$\rightarrow C_{pk}=min(C_{pki}, C_{pks})$$

$$\rightarrow C_{pk}=min(C_{pki}=\frac{\bar{X'}-LIE}{3\sigma'}, C_{pks}=\frac{LSE-\bar{X'}}{3\sigma'})$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación del proceso.

Por tanto, el índice Cpk es:

In [20]:
X = 370 # Se toma X' como el mejor estimador de mu

Cpki = np.round((X - LIE)/(3*s),2)
Cpks = np.round((LSE - X)/(3*s),2)

Cpk = np.min([Cpki, Cpks])

print("Cpki = "+ str(Cpki))
print("Cpks = "+ str(Cpks))
print("Cpk = "+ str(Cpk))

Cpki = 0.45
Cpks = 0.76
Cpk = 0.45


Dado que Cpk $\approx$ 0.45, esto confirma que el proceso no es realmente capaz y por lo menos tiende a no cumplir con una de las especificaciones de calidad. Dicho lo anterior, tal parece que no se está cumpliendo en mayor medida con el límite inferior de calidad, pues corresponde a Cpki $\approx$ 0.45. De igual forma, el proceso tampoco parece cumplir con el límite superior de especificación, ya que Cpks $\approx$ 0.76.

**c) Calcule los límites de fluctuación natural del proceso (LFNP):**

R/ Se tiene que los límites de fluctuación natural del proceso están dados por la siguiente fórmula:

$$\rightarrow LFN_{i,s} = \bar{X'} \pm 3\sigma'$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación estándar del proceso.

Entonces, los límites de fluctuación son:

In [21]:
LIFN = np.round(X - 3*s,4)
LSFN = np.round(X + 3*s,4)

print("LIFN = " + str(LIFN) + " y LSFN = " + str(LSFN))

LIFN = 337 y LSFN = 403


Se compara si LIFN es mayor o igual a LIE y si LSFN es menor o igual a LSE:

In [23]:
print(LIFN>=LIE, LSFN<=LSE)

False False


Se confirma que ninguno de los límites de fluctuación natural del proceso se encuentran dentro de las especificaciones de calidad.

**d) Dado que para contratar al operador es prioritario que la velocidad de transferencia cumpla con los requerimientos mínimos de calidad, y además, las velocidades de transferencia por encima de la espeficicación superior no representan un problema. Por el momento al CEO solo le interesa saber ¿Qué porcentaje de las conexiones no cumplen con el límite inferior de especificación?**

R/ Inicialmente se debe determinar el valor Z estandarizado para dicho límite (LIE). Esto se obtiene con la siguiente fórmula:

$$\rightarrow Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}$$

Obteniendo el siguiente resultado:

In [24]:
Zinf = np.round((LIE - X)/s,4)
print(Zinf)

-1.3636


In [25]:
P = np.round(st.norm.cdf(Zinf),4)
print(P)

0.0863


Es decir que la proporción de veces que el operador estaría por debajo del límite inferior de especificación es del 8.63%. El porcentaje es relativamente bajo, representarían aproximadamente 863 de cada 10.000 conexiones, pero se podría mejorar reduciendo un poco más la variabilidad del proceso.

**e) Suponiendo que se pudiera reducir drásticamente la variabilidad del proceso, ¿Qué valor máximo debería tener la desviación estándar para que a lo sumo 3% de las veces se incumpla con el límite inferior de especificación?**

Inicialmente se debe despejar $\sigma'$ de la fórmula de Z estandarizado inferior, así:

$$Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}\rightarrow\sigma'_{inf}=\frac{LIE-\bar{X'}}{Z_{inf}}$$

Donde:
* $Z_{inf}$ es el valor de Z estandarizado inferior que representa una probabilidad de 3.0% de obtener velocidades de transferencia por debajo de LIE.

Por lo tanto:

In [28]:
VO_inf = 0.03
Zinf_vo = np.round(st.norm.ppf(VO_inf),4)

print("Zinf = "+str(Zinf_vo))

Zinf = -1.8808


Entonces, los valores entre los cuales puede oscilar $\sigma'$ para que la velocidad no cumpla con las especificaciones en máximo 3.0% de las veces son:

In [29]:
s_inf_vo = np.round((LIE - X)/Zinf_vo,5) # Cuando el proceso genera velocidades por debajo del LIE el 3% de las veces.
print("Sinf = " + str(s_inf_vo))

Sinf = 7.97533


Por tanto, el proceso debe tener mejorar su precisión debido a que la desviación estándar debe ser máximo 7.97 Mbps para obtener a lo sumo 3% de velocidades por debajo del LIE.